src/cluster.c

```diff
@@ -4597,44 +4637,59 @@ NULL
         /* Check that all the arguments are parseable and that all the
          * slots are not already busy. */
         for (j = 2; j < c->argc; j++) {
-            if ((slot = getSlotOrReply(c,c->argv[j])) == -1) {
+            if ((slot = getSlotOrReply(c,c->argv[j])) == C_ERR) {
                 zfree(slots);
                 return;
             }
-            if (del && server.cluster->slots[slot] == NULL) {
-                addReplyErrorFormat(c,"Slot %d is already unassigned", slot);
+            if(getSlotStatus(c, slots, del, slot, slot) == C_ERR){
```

Comment:
A formatter would help here.
```suggestion
            if (getSlotStatus(c, slots, del, slot, slot) == C_ERR) {
```